content-watcher.js

```diff
@@ -34,31 +36,52 @@ function sendPing() {
 }
 
 function putUpdate(path) {
-  const data = JSON.stringify({
-    relative_path: path,
-    contents: fs.readFileSync(path).toString()
-  })
-  const options = postRequestOptions(
-    `/attendances/${process.env.ATTENDANCE_ID}/file_snapshots`,
-    {
-      "Content-Type": "application/json",
-      "Content-Length": data.length
-    }
-  )
+  if(!changed_files.includes(path)) {
+    changed_files.push(path)
+  }
+}
 
-  const req = https.request(options, res => {
-    console.log(`status: ${res.statusCode}`)
+function uploadChangedFiles() {
+  // Iterate over a copy so the indexes don't change as we modify the array
+  changed_files_copy = [...changed_files]
 
-    res.on("data", d => {
-      process.stdout.write(d)
+  changed_files_copy.forEach(function(path) {
+    process.stdout.write(`uploading: ${path}\n`)
+    const data = JSON.stringify({
+      relative_path: path,
+      contents: fs.readFileSync(path).toString()
     })
-  })
+    const options = postRequestOptions(
+      `/attendances/${process.env.ATTENDANCE_ID}/file_snapshots`,
+      {
+        "Content-Type": "application/json",
+        "Content-Length": data.length
+      }
+    )
+
+    const req = https.request(options, res => {
+      process.stdout.write(`status: ${res.statusCode}\n`)
 
-  req.on("error", error => {
-    console.error(error)
+      res.on("data", d => {
+        process.stdout.write(d)
+        process.stdout.write("\n")
+      })
+
+      // 1xx and 2xx status codes are successful for our purposes
+      if(res.statusCode < 300) {
+        this_index = changed_files.indexOf(path)
+        if(this_index >= 0) {
+          changed_files.splice(this_index, 1)
+        }
```

Comment:
I often wish Javascript could have better built-in methods...

Reply:
Goodness me yes.